### ISSUE TITLE
Docs: Bug in the code sample for server actions in client component.

### ISSUE THREAD

**Issue by iamrahulrnair** [vercel/next.js#53910]
### What is the improvement or update you wish to see?

![image](https://github.com/vercel/next.js/assets/55950758/a542405e-e33e-4739-b2b4-d45afaf2ab7a)
[Here](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions#with-client-components) it states that we cant define actions directly inside a client component. but the code does the exact opposite.

one possible fix is:- 
```
export async function action(propAction){
  await propAction
}
```
and import this in  app/components/example-client-component.js and change:-
```
 <form action={action(updateItem)}>
      <input type="text" name="name" />
      <button type="submit">Update Item</button>
    </form>
```
 If this looks good i can raise a pr for it.

### Is there any context that might help us understand?

Server actions when passing action as a prop to child client component

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions#props

**Comment by iamrahulrnair** [comment]
fix #53912, review?

**Comment by schimi-dev** [comment]
I am not sure if this is a bug. 

From how I understand this, the function named  `action` is actually a Client Action that internally calls the Server Action named `updateItem`. Then, the form receives the Client Action (not the Server Action directly) as `action` prop. 

Using Server Actions composed within Client Actions that way seems to work quite well from what I tested. Maybe it would be good to state in the docs whether it is an allowed pattern to wrap Server Action in Client Actions and then pass that Client Action as `action` prop to a form.

Technically, this patterns seems to work fine at the moment, and I think there might be a lot of use cases for it (e.g. Client side Form validation or preprocessing of data so that you do not have to pass FormData to the Server Action but some already preprocessed custom object).
